geth-patches/README.md

```diff
@@ -0,0 +1,85 @@
+# Status Patches to for geth (go-ethereum)
+---
+
+Status-go uses [go-ethereum](https://github.com/ethereum/go-ethereum) (**upstream**) as its dependency. As any other Go dependency `go-ethereum` code is vendored and stored in `vendor/` folder.
+
+However, there are a few changes has been made to the upstream, that are specific to Status and should not be merged to the upstream. We keep those changes as a set of patches, that can be applied upon each next release of `go-ethereum`. Patched version of `go-ethereum` is available in the [status-im/go-ethereum](https://github.com/status/go-ethereum) repo.
+
+We try to minimize number and amount of changes in those patches as much as possible, and whereas possible, to contribute changes into the upstream.
+
+# Patches
+
+ - `0000-accounts-hd-keys.patch` — adds support for HD extended keys (links/docs?)
+ - `0001-wnode-something.patch` – adds something to the wnode :) (need to check if it is needed at all)
+ - `0002-les-api-status.patch` — adds StatusBackend into LES code (need to be inspected, some things can and should be done outside of les code)
+ - `0003-dockerfiles-wnode-swarm.patch` — adds Dockerfiles (who uses this?)
+ - `0004-whisper-notifications.patch` — adds Whisper notifications (need to be reviewed and documented)
+ - `0005-whisper-delivery.patch` - adds support for logs/traces of Whisper traffic (questionable, nobody used this functionality so far)
+ - `0006-latest-cht.patch` – updates CHT root hashes, should be updated regularly to keep sync fast, until proper Trusted Checkpoint sync is not implemented as part of LES/2 protocol.
+ - `0007-README.patch` — update upstream README.md.
+
+# Updating upstream version
+
+When a new stable release of `go-ethereum` comes out, we need to upgrade our fork and vendored copy.
+
+**Note: The process is completely repeatable, so it's safe to remove current `go-ethereum` directory, clone latest upstream version and apply patches from scratch.**
+
+## How to update forked version
+Make sure you have `status-go` in your `$GOPATH/src/github.com/status-im/` first.
+
+### From very scratch
+Use this method if you're up to nuke forked repo for some reason and apply patches from scratch:
+
+```
+# from scratch
+rm -rf $GOPATH/src/github.com/status-im/go-ethereum
+cd $GOPATH/src/github.com/status-im/
+git clone https://github.com/ethereum/go-ethereum
+
+# update remote url to point to our fork repo
+ git remote set-url origin git@github.com:status-im/go-ethereum.git
```

Comment:
We could also recommend adding `git remote set-url upstream git@github.com:ethereum/go-ethereum.git` so any further pulls from upstream could be done like `git pull upstream release/1.7:master`.

Reply:
Great! Feel free to update README. I like the way you proposed — it's more explicit, but can't check it now from scratch, so let's update this after this PR is merged. Both options work, right?

Reply:
It sounds good. I will make a PR when this one is merged. Also, I will test the whole flow shortly after this PR is merged as I have a use-case for a new patch.